### ISSUE TITLE
Does L-BFGS support mini-batch gradient decent?

### ISSUE THREAD

**Issue by Cyberface** [JuliaNLSolvers/Optim.jl#875]
Hi,

I was wondering if the L-BFGS optimiser support mini-batch gradient decent when when training an artificial neural network? I ask because I don't think pytorch supports this natively.

Thanks!
Sebastian

**Comment by pkofod** [comment]
Not explicitly. We might be able to do something. Can you give me a small example?

**Comment by Cyberface** [comment]
Hi @pkofod,

Thanks for getting back to me. My interest was more of a passing interest to see if one could use this for larger datasets and I'm not really proficient in Julia! So i'll just close this issue I guess and sorry if I wasted your time!

**Comment by ChrisRackauckas** [comment]
It does through GalacticOptim.jl's data interface which uses the Optim callback to handle the data iterator updates.

**Comment by Cyberface** [comment]
Ah I see thanks!
